RecoTracker/LSTCore/src/alpaka/Quintuplet.h

```diff
@@ -1710,107 +1729,146 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
-
-            bool tightCutFlag = false;
-            bool success = runQuintupletDefaultAlgo(acc,
-                                                    modules,
-                                                    mds,
-                                                    segments,
-                                                    triplets,
-                                                    lowerModule1,
-                                                    lowerModule2,
-                                                    lowerModule3,
-                                                    lowerModule4,
-                                                    lowerModule5,
-                                                    innerTripletIndex,
-                                                    outerTripletIndex,
-                                                    innerRadius,
-                                                    outerRadius,
-                                                    bridgeRadius,
-                                                    regressionCenterX,
-                                                    regressionCenterY,
-                                                    regressionRadius,
-                                                    rzChiSquared,
-                                                    chiSquared,
-                                                    nonAnchorChiSquared,
-                                                    dBeta1,
-                                                    dBeta2,
-                                                    tightCutFlag,
-                                                    t5Embed,
-                                                    ptCut);
-
-            if (success) {
-              int totOccupancyQuintuplets = alpaka::atomicAdd(
-                  acc, &quintupletsOccupancy.totOccupancyQuintuplets()[lowerModule1], 1u, alpaka::hierarchy::Threads{});
-              if (totOccupancyQuintuplets >= ranges.quintupletModuleOccupancy()[lowerModule1]) {
+
+            unsigned int secondSegmentIndex = triplets.segmentIndices()[innerTripletIndex][1];
+            unsigned int thirdSegmentIndex = triplets.segmentIndices()[outerTripletIndex][0];
+
+            unsigned int innerOuterOuterMiniDoubletIndex =
+                segments.mdIndices()[secondSegmentIndex][1];  //inner triplet outer segment outer MD index
+            unsigned int outerInnerInnerMiniDoubletIndex =
+                segments.mdIndices()[thirdSegmentIndex][0];  //outer triplet inner segment inner MD index
+
+            //this cut reduces the number of candidates by a factor of 3, i.e., 2 out of 3 warps can end right here!
+            if (innerOuterOuterMiniDoubletIndex != outerInnerInnerMiniDoubletIndex)
+              continue;
+
+            // Match inner Sg and Outer Sg
+            int mIdx = alpaka::atomicAdd(acc, &matchCount, 1, alpaka::hierarchy::Blocks{});
```

Comment:
Similar comment below, I'm not sure why this atomic add needs to be over alpaka::hierarchy::Blocks{}? matchCount is in shared memory, which is shared within a given block so can't this by alpaka::hierarchy::Threads{}?

Reply:
You are right, I think this can be a hierarchy::Threads, since it doesn't need block level atomicAdd, I will add and test